pkg/utils/utils.go

```diff
@@ -948,3 +948,30 @@ func DeleteSvcSnapshot(svcSnapshotName string, gcSnapshotName string, gcSnapshot
 	return nil
 }
 
+func GetVcConfigSecretFilterFunc(logger logrus.FieldLogger) func(obj interface{}) bool {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		logger.WithError(err).Errorf("Failed to get k8s inClusterConfig")
+		return nil
+	}
+	var ns string
+	// Get the cluster flavor
+	clusterFlavor, err := GetClusterFlavor(config)
+	if clusterFlavor == constants.Supervisor {
+		ns = constants.VCSecretNsSupervisor
+	} else if clusterFlavor == constants.VSphere {
+		ns = constants.VCSecretNs
+	}
+	return func(obj interface{}) bool {
+		switch obj.(type) {
+		case  *k8sv1.Secret:
+			incomingSecret := obj.(*k8sv1.Secret)
+			return incomingSecret.Namespace == ns &&
+				incomingSecret.Name == constants.VCSecret
+		default:
+			logger.Debugf("Unrecognized object type %+v found during filtering, ignoring", obj)
```

Comment:
Don't print out this object in the logs.  It may contain credentials, etc.  We are not sure about it.